cumulusci/utils/options.py

```diff
@@ -18,6 +19,28 @@ def _describe_field(field):
     return rc
 
 
+class ReadOnlyOptions(dict):
+    """To enforce self.options to be read-only"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def __setitem__(self, key, value):
+        raise TaskOptionsError(
+            "The 'options' dictionary is read-only. Please use 'parsed_options' instead."
```

Comment:
±: How about extracting this error message into a module-level constant? Doing so would apply the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle and be consistent with the codebase.

Reply:
Yes will make this change